server/src/main/java/org/elasticsearch/cluster/metadata/ProjectMetadata.java

```diff
@@ -106,6 +110,16 @@ public class ProjectMetadata implements Iterable<IndexMetadata>, Diffable<Projec
 
     private final IndexVersion oldestIndexVersion;
 
+    public static final ClusterBlock PROJECT_UNDER_DELETION_BLOCK = new ClusterBlock(
```

Comment:
This is the only project-global block. Most likely there will be another one later called `PROJECT_RECOVERING_BLOCK` used during resurrection (or also creation). The only difference being that one would be retryable but this one is not.

Reply:
Sure we can go without it first.